apps/dashboard/app/models/user_configuration.rb

```diff
@@ -54,6 +57,11 @@ class UserConfiguration
     ConfigurationProperty.property(name: :custom_css_files, default_value: []),
 
     ConfigurationProperty.property(name: :dashboard_title, default_value: 'Open OnDemand', read_from_env: true),
+
+    # Navigation properties
+    ConfigurationProperty.with_boolean_mapper(name: :show_all_apps_link, default_value: false, read_from_env: true, env_names: ['SHOW_ALL_APPS_LINK']),
+    ConfigurationProperty.with_boolean_mapper(name: :nav_categories_filter, default_value: NavConfig.categories_whitelist?),
```

Comment:
I don't believe we need `nav_categories_filter`. We have `NavConfig.categories_whitelist?` because the code cannot determine whether `NavConfig.categories` was supplied by us (the default) or the overwritten by the user because it's an `attr_accessor`.

With the the configuration `nav_categories` we can make this determination if the default is nil. Making the allowlist implicit. I.e., If I've defined nav_categories -> then those are the categories allowed.

It'll add complexity on our side, but it's simplicity on theirs - you don't need to set 2 things to do 1 thing.

Reply:
Initially I thought the same, but looking at the implementation in `ApplicationController.filter_groups`, I think we do.

The `NavConfig.categories` are used to order the nav items, when `NavConfig.categories_whitelist?` is true, it filters as well as orders. Using only the categories and setting it to `nil` to avoid filter will break the current ordering logic.

Reply:
Yea 😆 I went to mock up the logic and found it would be a bit complicated.

Maybe nav_categories_filter (or filter_nav_categories?) is internal & computed?

```ruby
  def filter_nav_categories?
    fetch(:categories, nil) || NavConfig.categories_whitelist?
  end

  def categories
    fetch(:categories, nil) || NavConfig.categories
  end 
```

Reply:
I am not sure you can achieve the same behaviour with a single property.

The above will force filtering when categories are defined. Currently you can change categories and only change the order.

Reply:
> Currently you can change categories and only change the order.

I don't follow this exactly.

I've got to look up the history here.

* if you do nothing, you have the same behavior as always.
* the upgrade path is the same for folks who use whitelist and those who don't. 
* Those who don't use whitelists, but configured the categories, wont' have auto-discovery.  But they made a conscious choice to configure the categories, so what's the expectation here? 
  *  They choose to configure categories but then also want the additional categories that _may_ come up? 

I guess my point is, if you've configured some but not all categories do you still expect to auto discover the others?  It feels a little like half configured to me. or that folks do this and it works by coincidence.

I guess I'm unclear about user expectation when you make a conscious choice to configure a thing (provide some but not all categories) but then allow for auto discovery of new groups.

Reply:
My initial objective was to support the same functionality, but using profile based configuration for the same properties.

> Currently you can change categories and only change the order.

This is the current behaviour when whitelist is false. The auto-discovery still works, all default nav items are appended to the nav categories in alphabetical order.
```
categories_whitelist = false
categories = ["Apps", "Files", "Jobs", "Clusters", "Interactive Apps"]
Navigation:
Files, Jobs, Clusters, Interactive Apps, My Interactive Sessions

categories_whitelist = false
categories = ["Clusters", "Interactive Apps"]
Navigation:
Clusters, Interactive Apps, Files, Jobs, My Interactive Sessions

categories_whitelist = false
categories = ["Aday"]
Navigation:
Clusters, Files, Interactive Apps, Jobs, My Interactive Sessions
```

I don't think we can achieve this behaviour with a single property.

Reply:
This is the case that seems off to me. Defining half the order and leaving the rest. 
```
categories_whitelist = false
categories = ["Clusters", "Interactive Apps"]
Navigation:
Clusters, Interactive Apps, Files, Jobs, My Interactive Sessions
```

I guess that's my sticking point. That there's something implicitly happening, where I'd kind of rather just be explicit. 

Give me today to actually confirm this (and lookup the provenience for this feature), but I believe I'm OK with removing auto discovery in this case. Again, we'll have backward compatibility - so doing nothing means it'll work just as it always has.

But if you _do_ want to update & change the navbar, changing
`categories = ["Clusters", "Interactive Apps"]` (in ruby)
to 
`nav_categories_filter = ["Clusters", "Interactive Apps", "Files", "Jobs"]` (in yaml)
I don't think is a stretch.

Reply:
OK looked at the history a bit and dug up https://github.com/OSC/ood-dashboard/issues/295 and https://github.com/OSC/ood-dashboard/pull/385.  It would appear that I'm suggesting we actually revert back to 1.3 behavior.

Looking at the docs - https://osc.github.io/ood-documentation/latest/customization.html#control-which-apps-appear-in-the-dashboard-navbar - there are _even more_ variants than we're discussing here (getting them to appear alphabetically) and that just solidified my position.

When we need 2+ paragraphs _and_ a table to describe what I feel is a fairly simple feature - that's an issue for me.

That section would then collapse to:
* do nothing and these groups [...] + any other category we discover will be on the navigation bar
* `nav_categories_filter` to specify the groups you want.

Everything's a trade-off I think we're making this trade:
Simplicity of configuration for everyone vs sites that has partial categories configured and expect discovery for the rest. 

I think I'm happy to make that trade because the latter I believe is in the minority perhaps even just 1-4 sites.

Reply:
So my vote is to remove it, to favor simplicity over what I imagine is an edge case or at least removing something that a very small subset of folks rely on.

Reply:
OK - Implementing the changes....

Reply:
Sorry if that took a bit! But seeing that table in the documentation kinda solidified it for me.

Reply:
I have completed the changes, ready for another review 🚀